dispatch/apps/events/views.py

```diff
@@ -13,16 +13,39 @@ def submit_success(request):
     return render(request, 'events/submit/success.html')
 
 def submit_form(request):
-    facebook_url = request.POST.get('facebook_url')
-    facebook_error = False
+    event_url = request.POST.get('event_url')
+    url_error = False
+
+    if request.POST.get('url_import') and event_url is not None:
+
+        handlers = {
+            'calendar.events.ubc.ca': {
+                'url_handler': UBCEvent,
+                'url_handler_error': UBCEventError
+            },
+            'facebook.com/events/': {
```

Comment:
I really like the approach of using URL prefixes to match handlers, but I think it might be even better to use the domain name itself.

Can you try using regex to parse the hostname from the URL? In this case, the two valid hostnames would be:

```py
'calendar.events.ubc.ca'
'facebook.com'
```

Then you can define a simple lookup table:

```py
sources = {
  'calendar.events.ubc.ca': UBCEvent,
  'facebook.com': FacebookEvent
}

hostname = get_host_from_url_with_regex(url)

if hostname in sources:
  handler = sources[hostname]
  # ...
else:
  url_error = True